include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/LLVMFlowFunctions.h

```diff
@@ -10,413 +10,424 @@
+                  Alloc->getAllocatedType()->getArrayNumElements() > 0 &&
+                  Alloc->getAllocatedType()
+                      ->getArrayElementType()
+                      ->isStructTy() &&
+                  Alloc->getAllocatedType()
                           ->getArrayElementType()
-                          ->isStructTy() &&
-                      Alloc->getAllocatedType()
-                              ->getArrayElementType()
-                              ->getStructName() == "struct.__va_list_tag") {
-                    Res.insert(Alloc);
-                  }
-                }
+                          ->getStructName() == "struct.__va_list_tag") {
+                Res.insert(Alloc);
               }
             }
-          } else {
-            assert(Idx < Formals.size() &&
-                   "Out of bound access to formal parameters!");
-            if (FormalPredicate(Formals[Idx])) {
-              Res.insert(Formals[Idx]); // corresponding formal
-            }
           }
         }
       }
+
+      return Res;
     }
-    // Handle ordinary case
-    // Map actual parameters to corresponding formal parameters.
-    for (unsigned Idx = 0; Idx < Actuals.size() && Idx < DestFun->arg_size();
-         ++Idx) {
-      if (Source == Actuals[Idx] && ActualPredicate(Actuals[Idx])) {
-        assert(Idx < Formals.size() &&
-               "Out of bound access to formal parameters!");
-        Res.insert(Formals[Idx]); // corresponding formal
+
+    llvm::PointerIntPair<const llvm::CallBase *, 1, bool> CSAndPropGlob;
+    llvm::PointerIntPair<const llvm::Function *, 1, bool> DestFunAndPropZero;
+    std::decay_t<Fn> PropArg;
+  };
+
+  return std::make_shared<Mapper>(
+      CallSite, DestFun, PropagateGlobals, PropagateZeroToCallee,
+      std::forward<Fn>(PropagateArgumentWithSource));
+}
+
+/// A flow function that serves as default-implementation of the return flow
+/// function. For more information about return flow functions, see
+/// FlowFunctions::getRetFlowFunction().
+///
+/// Propagates the return value back to the call-site and based on the
+/// PropagateParameter predicate propagates back parameters holding as dataflow
+/// facts.
+///
+/// Let a call-site cs: r = fun(..., ax, ...) a function prototype fun(...,
+/// px, ...) and an exit statement exit: return rv.
+/// Further given a flow function f = mapFactsToCaller(cs, exit, ...). Then for
+/// all dataflow facts x holding at exit:
+///   f(rv) = {r}   if PropagateParameter(rv, rv) else {},
+///   f(px) = {ax}  if PropagateParameter(px, px) else {},
+///   f(g)  = {g}   if PropagateGlobals else {},
+///   f(Λ)  = {Λ}   if PropagateZeroToCaller else {},
+///   f(x)  = {ax}  if PropagateParameter(ax, x) else {}.
+///
+/// NOTE: PropagateParameter is used to dermine if both output parameters and
+/// the return value should be propagated back to the caller. So, it may make
+/// sense to provide two different predicates combined by psr::Overloaded.
```

Comment:
I think it might make the analysis code even more readable and this function easier to use if we explicitly expose two functions one for each case, instead of this general form.

Reply:
done